patches/tModLoader/Terraria/ModLoader/ModHair.cs

```diff
@@ -0,0 +1,53 @@
+using System.Linq;
+using System.Collections.Generic;
+using Microsoft.Xna.Framework.Graphics;
+using Terraria.GameContent;
+
+namespace Terraria.ModLoader;
+
+/// <summary>
+/// This class allows you to define custom hair styles for a player, controlling the associated gender and unlock conditions.
+/// </summary>
+public abstract class ModHair : ModTexturedType
+{
+	public int Type { get; internal set; }
+
+	/// <summary>
+	/// The path to the alternative texture used for when the hair is covered by a hat.
+	/// </summary>
+	public virtual string AltTexture => Texture + "_Alt";
+
+	/// <summary>
+	/// Determines whether this hairstyle is considered male, which influences what gender is picked when randomizing a character.
+	/// </summary>
+	public virtual bool IsMale => true;
```

Comment:
Male bias here is not preferable (yes, Terraria isn't super good about this thing). Suggest:
```cs
	enum Gender { Unspecified, Male, Female }

	/// <summary>
	/// Used to set the character gender based on hair style when randomizing a new character. <br/>
	/// If Unspecified, the gender will be randomly rolled. <br/>
	/// Note that all hair styles can be selected with either gender. This is just a default for quick randomization
	/// </summary>
	public virtual Gender TypicalCharacterGender => Gender.Unspecified;
```

Reply:
Not sure this is a great idea, it becomes inconsistent with ``isMale`` on Player.

Reply:
or perhaps we should just go with a name which represents what it is actually used for such as ``createsMaleCharacterOnRandomization``

Reply:
How would it be inconsistent?